api/clients/v2/codecs/blob.go

```diff
@@ -27,6 +28,13 @@ type Blob struct {
 
 // DeserializeBlob initializes a Blob from bytes
 func DeserializeBlob(bytes []byte, blobLengthSymbols uint32) (*Blob, error) {
```

Comment:
Might be good to document why blobLengthSymbols is needed. We've had this discussion 2-3 times already and I still can't remember the exact reason :\

Reply:
This is documented on the `blobLengthSymbols` field of the struct:

```
	// blobLengthSymbols must be a power of 2, and should match the blobLength claimed in the BlobCommitment
	//
	// This value must be specified, rather than computed from the length of the coeffPolynomial, due to an edge case
	// illustrated by the following example: imagine a user disperses a very small blob, only 64 bytes, and the last 40
	// bytes are trailing zeros. When a different user fetches the blob from a relay, it's possible that the relay could
	// truncate the trailing zeros. If we were to say that blobLengthSymbols = nextPowerOf2(len(coeffPolynomial)), then the
	// user fetching and reconstructing this blob would determine that the blob length is 1 symbol, when it's actually 2.
```